### PR TITLE
chore(automation): extract personal Claude Code harness to the private reeve layer

### PR DESCRIPTION
**Pull request by joshsmithxrm**
## Summary

Extract the maintainer's personal Claude Code automation out of this public repo, keeping only project-knowledge skills and damage-prevention hooks. **No product code changes** — this is dev-tooling only.

The concern that prompted this: the checked-in `.claude/settings.json` wired 22 hooks that bind *every* Claude Code session in the repo with no per-user opt-out — including a `session-stop-workflow` gate that refused to end a session until `/gates` → `/verify` → `/pr` completed, and a `debug-first` gate that blocked fix attempts. Those enforce a multi-agent, high-token pipeline that's reasonable for the maintainer's budget but unreasonable to impose on a contributor (e.g. on a Pro plan). The skills themselves were never forced on anyone — but the hooks were.

**Principle applied:** the repo owns facts, safety, and outcome contracts; the person owns process, enforcement, and spend. CI remains the single authoritative gate — it already runs the same `dotnet`/`npm` commands the local skills did — so a non-Claude contributor is completely unaffected.

## What changed

**Removed (personal enforcement/process tier):**
- De-bound and deleted the 12 process-enforcement hooks: `session-start-workflow`, `session-stop-workflow`, `debug-first`, `pr-gate`, `pre-commit-validate`, `post-commit-state`, `inflight-auto-deregister`, `review-guard`, `retro-html-guard`, `stop-hook-watchdog`, `checkout-guard`, `preuse-askuserquestion` (+ unwired `notify`).
- Deleted 10 personal-workflow skills: `start`, `cleanup`, `status`, `orchestrate`, `retro`, `investigate`, `await-operator`, `resume-with-answer`, `recover-session`, `write-agent`.
- Deleted the orchestration machinery (`pipeline.py`, `claude_dispatch.py`, `pr_monitor.py`, goal supervisor/loop, spawn/inflight/triage scripts) with their specs and tests, tracked `.retros/` history, and `interaction-patterns.md`.

**Kept (project-knowledge + safety tier):**
- 19 knowledge skills (`setup`, `release`, `test-conventions`, `ext-panels`, `gates`, `verify`, `qa`, `review`, `pr`, `spec`, `spec-audit`, `design`, `implement`, `backlog`, `dependabot-triage`, `shakedown`, `audit-capture`, `debug`, `write-skill`) — each stripped of workflow-state/pipeline tendrils so it stands alone as documentation-grade procedure. All 6 agents kept.
- 10 safety/governance hooks stay wired: `protect-main-branch`, `snk-protect`, `rm-guard`, `worktree-safety`, `shakedown-safety` (+ `shakedown-readonly-guard`), `settings-json-guard`, `claudemd-line-cap`, `skill-line-cap`, `sdk-spend-warn`.

**Edited:** `CLAUDE.md` (machinery directives removed; `/gates`→`/verify`→`/pr` and `/debug` are now advisory, not hook-enforced), `CONTRIBUTING.md` (new "AI-Assisted Development (Optional)" section — the tooling is optional, CI is the contract, with a token-cost note).

The two enforcement behaviors worth keeping (a debug-first nudge and an unshipped-work stop-gate) move to the maintainer's private reeve plugin as generic, opt-in, repo-agnostic hooks — see the companion PR joshsmithxrm/reeve#1431. The retired pipeline is captured as a reeve backlog item, not rebuilt here.

## Test plan

- `pytest tests/test_hook_paths.py tests/test_hooks.py` (the `workflow-tests.yml` CI job) — **38 passed**.
- Hook-envelope, skill-content, and bash-portability suites — **29 passed**.
- Full suite: **508 passed, 6 failed** — the 6 failures are `test_release_skill_content.py`, **pre-existing on main** (those release sections are absent from `release/SKILL.md` at HEAD, unrelated to this change) and not run by CI's Python job. Tracked separately.
- `.claude/settings.json` validated: exactly the 10 safety/governance hooks remain wired, 0 enforcement hooks leaked.
- No `src/` changes → .NET/TS build and unit gates are unaffected (CI will confirm).

## Notes for reviewers

- Deletion-heavy (154 files, ~48k lines removed). `[size-waived: automation-extraction]` — a coherent single-purpose extraction that cannot be meaningfully split below the size gate without leaving the repo in a half-migrated state.
- The methodology isn't lost — it remains in git history, and the reeve absorption issue points at the pre-extraction SHA.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

[size-waived: automation-extraction]

## Summary by CodeRabbit

* **Documentation**
  * Updated shakedown environment-variable contract guidance and tightened writer rules.
  * Revised skill/setup documentation for design, implement, QA, review, gates, and PR lifecycle (including automated review comment handling).
  * Removed obsolete workflow/orchestration/retro/notification/session-management documentation and tightened portable Bash guidance.
* **Breaking Changes**
  * Removed many local guard/notification/session enforcement behaviors, so previous workflow “guardrails” no longer run.
* **Chores**
  * Simplified test/verification coverage by removing multiple related hook/workflow/retro test suites and helper tooling.

### CODE REVIEW

File: .claude/skills/pr/SKILL.md [hunk 1]
```diff
@@ -57,41 +38,49 @@ Write body to temp file (see REFERENCE.md §3 for template), then:
...
-```
+### Step 5: Wait for Gemini Review and CI
 
-### Step 5: Launch Background Monitor (MANDATORY) <!-- enforcement: T1 hook:session-stop-workflow -->
+Poll for the automated Gemini review and check status directly:
 
 ```bash
-python scripts/pr_monitor.py --worktree "$(pwd)" --pr {pr-number}
+gh pr checks {N} --watch          # CI + CodeQL status
+gh pr view {N} --json reviews,comments,statusCheckRollup
```
[the remark this answers is not in the document]
Comment (joshsmithxrm) [reply .claude/skills/pr/SKILL.md]: Fixed in 5874697d9 — standardized the PR-number placeholder to `<pr-number>` in Step 5.

File: .claude/skills/pr/SKILL.md [hunk 2]
```diff
@@ -57,41 +38,49 @@ Write body to temp file (see REFERENCE.md §3 for template), then:
...
+
+### Step 6: Triage Every Comment
+
+Fetch review comments and respond to **every** one — this is the discipline that captures knowledge:
+
 ```bash
-cat .workflow/pr-monitor.pid
-python scripts/workflow-state.py set pr.monitor_launched now
+gh pr view {N} --json reviews,comments
+gh api repos/{owner}/{repo}/pulls/{N}/comments      # inline review comments
```
[the remark this answers is not in the document]
Comment (joshsmithxrm) [reply .claude/skills/pr/SKILL.md]: Fixed in 5874697d9 — switched to the native `:owner`/`:repo` placeholders and `<pr-number>` in Step 6.

File: .claude/skills/pr/SKILL.md [hunk 3]
```diff
@@ -57,41 +38,49 @@ Write body to temp file (see REFERENCE.md §3 for template), then:
...
 
-1. **Monitor**: confirm `.workflow/pr-monitor.pid` exists and process running (`kill -0`). If missing AND no fallback recorded → fail: `"⚠ Monitor PID file missing"`.
-2. **Gemini review**: poll `gh pr view {N} --json reviews,comments` every 30s for 5 min. If absent → fail. Bypass with `--skip-gemini-check`.
+### Step 7: Flip to Ready and Present Summary
 
-### Step 7: Present Summary
+Once CI is green, the Gemini review is triaged, and every comment is answered:
 
+```bash
+gh pr ready {N}
```
[the remark this answers is not in the document]
Comment (joshsmithxrm) [reply .claude/skills/pr/SKILL.md]: Fixed in 5874697d9 — `gh pr ready <pr-number>`.

File: .claude/skills/pr/REFERENCE.md [hunk 4]
```diff
@@ -48,33 +36,30 @@ Closes #NNN
...
-Launch command: `python scripts/pr_monitor.py --worktree "$(pwd)" --pr {pr-number}`
-
-- **Windows:** `subprocess.Popen(..., creationflags=subprocess.CREATE_BREAKAWAY_FROM_JOB | subprocess.CREATE_NEW_PROCESS_GROUP)`
-- **Unix:** `subprocess.Popen(..., start_new_session=True)`
-
-Fallback (monitor can't launch): wait inline, triage yourself, convert to ready. Record reason:
 ```bash
-python scripts/workflow-state.py set pr.monitor_launched "fallback: <reason>"
+gh pr checks {N} --watch                          # blocks until CI + CodeQL settle
+gh pr view {N} --json reviews,comments            # poll for the Gemini review
```
[the remark this answers is not in the document]
Comment (joshsmithxrm) [reply .claude/skills/pr/REFERENCE.md]: Fixed in 5874697d9 — standardized `<pr-number>` in REFERENCE §4.

File: .claude/skills/pr/REFERENCE.md [hunk 5]
```diff
@@ -48,33 +36,30 @@ Closes #NNN
...
 ```
 
+Poll every ~30s until the Gemini review appears (up to a few minutes). Late-arriving inline comments show up under `gh api repos/{owner}/{repo}/pulls/{N}/comments` — re-fetch before flipping to ready so nothing goes untriaged.
```
[the remark this answers is not in the document]
Comment (joshsmithxrm) [reply .claude/skills/pr/REFERENCE.md]: Fixed in 5874697d9 — native `:owner`/`:repo` plus `<pr-number>` in REFERENCE §4.

File: .claude/skills/pr/REFERENCE.md [hunk 6]
```diff
@@ -48,33 +36,30 @@ Closes #NNN
...
 
-The monitor writes final status to `.workflow/pr-monitor.log` on terminal state (MERGED/CLOSED). It does NOT auto-invoke `/cleanup`.
+```bash
+git worktree remove <path>        # if working in a dedicated worktree
+git branch -d <branch>
+git push origin --delete <branch> # if the remote branch was not auto-deleted
+```
 
-Future: opt-in `--cleanup-on-merge` flag with merge poller + single confirmation in final notification. Track via separate issue + spec.
+Confirm the merge first with `gh pr view {N} --json state` (expect `MERGED`).
```
[the remark this answers is not in the document]
Comment (joshsmithxrm) [reply .claude/skills/pr/REFERENCE.md]: Fixed in 5874697d9 — standardized `<pr-number>` in REFERENCE §5, including one further occurrence in the post-merge snippet.

File: .claude/skills/design/SKILL.md [hunk 7]
```diff
@@ -122,29 +93,19 @@ Co-Authored-By: {use the format from the system prompt}"
...
-     → python scripts/pipeline.py --worktree <cwd> --spec specs/<name>.md --from implement
-
-  2. Continue interactively
+  1. Continue interactively
      → /implement
 
-  3. Defer (pick up later)
+  2. Defer (pick up later)
      → Spec is committed on branch feat/<name>. Resume anytime.
 ```
```
[the remark this answers is not in the document]
Comment (joshsmithxrm) [reply .claude/skills/design/SKILL.md]: Fixed in 32b69f2a2 — added a `text` language to the handoff fence (MD040).

File: .claude/skills/design/SKILL.md [hunk 8]
```diff
@@ -122,29 +93,19 @@ Co-Authored-By: {use the format from the system prompt}"
...
-  1. Launch headless pipeline (recommended)
-     → python scripts/pipeline.py --worktree <cwd> --spec specs/<name>.md --from implement
-
-  2. Continue interactively
+  1. Continue interactively
      → /implement
 
-  3. Defer (pick up later)
+  2. Defer (pick up later)
      → Spec is committed on branch feat/<name>. Resume anytime.
```
[the remark this answers is not in the document]
Comment (joshsmithxrm) [reply .claude/skills/design/SKILL.md]: Fixed in 32b69f2a2 — reworded to 'the current feature branch'; no branch-naming convention is implied.

File: .claude/skills/implement/REFERENCE.md [hunk 9]
```diff
@@ -2,20 +2,13 @@
...
 ## §1 - No-Spec Fallback Chain (SKILL.md Input step)
 
-When `$ARGUMENTS` is absent and no spec is found via workflow state:
+When `$ARGUMENTS` is absent and no spec is found on the branch:
 
-1. **Check `.plans/context.md`** (written by `/start` from the issue body): if it exists, read it, generate a plan from the issue context and constraints it contains, save to `.plans/`, proceed.
+1. **Check `.plans/context.md`** (if present, e.g. written from an issue body): read it, generate a plan from the issue context and constraints it contains, save to `.plans/`, proceed.
 2. **No context.md**: prompt the user — "(1) Run `/design` to create a spec, or (2) describe the work for plan generation."
 3. **User chooses `/design`**: stop; instruct user to run `/design` first.
 4. **User chooses continue**: ask user to describe the work; generate a plan from their description.
```
[the remark this answers is not in the document]
Comment (joshsmithxrm) [reply .claude/skills/implement/REFERENCE.md]: Fixed in 32b69f2a2 — the no-spec fallback now reads specs/CONSTITUTION.md before generating any plan.

File: .claude/skills/implement/SKILL.md [hunk 10]
```diff
@@ -8,17 +8,15 @@ description: Execute a checked-in implementation plan end-to-end using parallel
...
 
 Agent tool, `/review`, `/verify`, `/qa`, `/debug`, `/gates`.
 
 ## Input
 
-`$ARGUMENTS` = path to plan file. If omitted: use spec from `.workflow/state.json`, generate plan, save to `.plans/`, proceed.
+`$ARGUMENTS` = path to plan file. If omitted: use the most relevant spec on the branch, generate a plan, save to `.plans/`, proceed.
 
-**Fallback — no spec:** check `.plans/context.md` first (written by `/start`); if present, generate plan from it. Otherwise: prompt user — run `/design` or continue without spec? See REFERENCE.md §1 for the full fallback chain.
+**Fallback — no spec:** check `.plans/context.md` first; if present, generate a plan from it. Otherwise: prompt the user — run `/design` or continue without spec? See REFERENCE.md §1 for the full fallback chain.
```
[the remark this answers is not in the document]
Comment (joshsmithxrm) [reply .claude/skills/implement/SKILL.md]: Fixed in 32b69f2a2 — the invocation examples and the $ARGUMENTS contract now align: a plan path, or a spec path that generates the plan first.

File: .claude/skills/implement/SKILL.md [hunk 11]
```diff
@@ -50,37 +39,26 @@ Read REFERENCE.md §3 for Opus vs. Sonnet guidance.
...
-### Step 6: Mandatory Tail — Full Verification Pipeline
+### Step 6: Mandatory Tail — Full Verification Sequence
 
 **A. Gates** — `/gates`
 **B. Verify** — `/verify extension|tui|cli|mcp` per changed surfaces
 **C. QA** — `/qa extension|cli|mcp|tui` per changed surfaces
 **D. Review** — `/review` final comprehensive review
 **E. Converge** — if critical/important findings: gates→review→fix loop (max 5 cycles)
-**F. Final State Check** — git log clean; `.workflow/state.json` timestamps all post-`started`
+**F. Final State Check** — git log clean; all phases committed.
```
[the remark this answers is not in the document]
Comment (joshsmithxrm) [reply .claude/skills/implement/SKILL.md]: Fixed in 32b69f2a2 — the final state check now requires `git status --porcelain` to produce no output.

File: .claude/skills/pr/REFERENCE.md [hunk 12]
```diff
@@ -48,35 +36,30 @@ Closes #NNN
...
+Poll every ~30s until the review appears (up to a few minutes); if no reviewer is configured, note that and proceed. Late-arriving inline comments show up under `gh api repos/:owner/:repo/pulls/<pr-number>/comments` — re-fetch before flipping to ready so nothing goes untriaged.
 
-> **Retro-enforced (PR #868):** Agent skipped the monitor, manually replied to 3 of 9 comments via `gh api`, missed all CodeQL comments. Manual comment triage is never an acceptable substitute.
+> **Lesson (PR #868):** an agent replied to only 3 of 9 comments and missed all CodeQL comments. Respond to **every** comment — fix the code or reply with a rationale. Partial triage is never acceptable.
```
[the remark this answers is not in the document]
Comment (joshsmithxrm) [reply .claude/skills/pr/REFERENCE.md]: Fixed in 32b69f2a2 — §4 now specifies polling every ~30s for up to 15 minutes, with explicit user confirmation required at the cap; the wait is skipped only when no reviewer app is installed.

File: .claude/skills/pr/SKILL.md [hunk 13]
```diff
@@ -57,49 +38,49 @@ Write body to temp file (see REFERENCE.md §3 for template), then:
...
 ```bash
-REVIEWER=$(python scripts/pr_monitor.py --print-reviewer)
-python scripts/workflow-state.py set pr.reviewer "$REVIEWER"
-python scripts/pr_monitor.py --worktree "$(pwd)" --pr {pr-number} --reviewer "$REVIEWER"
+gh pr checks <pr-number> --watch          # CI + CodeQL status
+gh pr view <pr-number> --json reviews,comments,statusCheckRollup
 ```
 
-Reviewer mode resolves `--reviewer` flag > `PPDS_PR_REVIEWER` env > repo default; `none` disables the external-review wait and triage. Launch as detached background process (see REFERENCE.md §4). Then:
+Review timing is unpredictable (2–10+ minutes). Poll `gh pr view <pr-number> --json reviews,comments` every ~30s until the review appears, up to a few minutes. If no reviewer is configured or it never arrives, note that and proceed — do not block indefinitely.
```
[the remark this answers is not in the document]
Comment (joshsmithxrm) [reply .claude/skills/pr/SKILL.md]: Fixed in 32b69f2a2 — Step 5 now polls up to 15 minutes and requires user confirmation before proceeding at the cap; never silently ready while a late review can land.

File: .claude/skills/qa/SKILL.md [hunk 14]
```diff
@@ -473,14 +443,8 @@ After fixing any issue found during verification:
...
+## QA Findings
 
-After each finding (PASS or FAIL), write findings to workflow state:
-
-```bash
-python scripts/workflow-state.py set qa_findings "<JSON array of findings>"
-```
+Record each finding (PASS or FAIL) as a structured entry so the review stage can deduplicate — review will skip findings already caught and fixed by QA.
 
 Each finding entry: `{"id": "QA-N", "surface": "ext|tui|mcp|cli", "description": "...", "file": "...", "severity": "critical|important|suggestion", "fixed": true|false, "fix_commit": "<sha>|null"}`
```
[the remark this answers is not in the document]
Comment (joshsmithxrm) [reply .claude/skills/qa/SKILL.md]: Fixed in 32b69f2a2 — findings are written to `.qa-findings.json` at the repo root (untracked, overwritten each /qa run) and /review Step 2b reads that artifact; both skills updated consistently.

File: .claude/skills/review/SKILL.md [hunk 15]
```diff
@@ -42,13 +42,7 @@ If $ARGUMENTS specifies a scope, filter the diff to those paths only.
...
 ### Step 2b: Load QA Findings for Dedup
 
-Before dispatching reviewers, read existing QA findings from state:
-
-```bash
-python scripts/workflow-state.py get qa_findings
-```
-
-Parse the output as JSON. Pass these findings to each reviewer subagent as "already found by QA" context. Reviewers should NOT re-report QA findings that were fixed (`fixed: true`) unless the fix introduced a new problem.
+If `/qa` ran earlier and produced findings, pass them to each reviewer subagent as "already found by QA" context. Reviewers should NOT re-report QA findings that were fixed (`fixed: true`) unless the fix introduced a new problem.
```
[the remark this answers is not in the document]
Comment (joshsmithxrm) [reply .claude/skills/review/SKILL.md]: Fixed in 32b69f2a2 — QA findings are now explicitly classified as permitted reviewer metadata used only for deduplication (the sanctioned exception to Step 3's isolation rule), and Step 3 names the appendix.

File: .claude/skills/setup/SKILL.md [hunk 16]
```diff
@@ -136,15 +136,15 @@ Skip on Linux — not meaningful in dev containers.
...
 ```bash
-python .claude/hooks/pr-gate.py < /dev/null
+echo '{}' | python .claude/hooks/protect-main-branch.py
 ```
 
-If Python is not available, warn that workflow hooks require Python 3. The hooks enforce workflow compliance (gates, verification, QA, review) before PR creation.
+If Python is not available, warn that the `.claude/` hooks require Python 3.
 
-Check that `.claude/settings.json` contains the hook configuration (SessionStart, PreToolUse, PostToolUse, Stop hooks).
+Check that `.claude/settings.json` contains the hook configuration (PreToolUse, PostToolUse, and any other configured hook events).
```
[the remark this answers is not in the document]
Comment (joshsmithxrm) [reply .claude/skills/setup/SKILL.md]: Fixed in 32b69f2a2 — the probe now pipes a valid nested envelope with an always-permitted `.claude/worktrees/` path and asserts exit 0.

File: .claude/skills/TWO-FILE-PATTERN.md [hunk 17]
```diff
@@ -96,10 +95,9 @@ keep the order even if you add new sections at the end.
   reproducible builds, macOS notarization)
 - §6 - NEVER list (with rationale for each)
 
-The same shape applies to `backlog` and `retro`: the SKILL.md
-runs the dispatch / phase loop; REFERENCE.md holds the label
-taxonomy, finding tiers, rule-change templates, and HTML artifact
-spec.
+The same shape applies to `backlog`: the SKILL.md runs the
+triage procedure; REFERENCE.md holds the label taxonomy and
+worked examples.
```
[the remark this answers is not in the document]
Comment (joshsmithxrm) [reply .claude/skills/TWO-FILE-PATTERN.md]: Fixed in 32b69f2a2 — the description now reads 'label taxonomy and procedural rules', matching the file's contents.

File: .claude/skills/verify/SKILL.md [hunk 18]
```diff
@@ -72,12 +72,12 @@ Per tool: valid->success shape; edge case->error handling; matches schema.
...
-pytest tests/test_pipeline.py tests/test_workflow_state.py tests/test_protect_main_branch.py tests/test_session_stop_workflow.py -v
+pytest tests/test_protect_main_branch.py tests/test_hooks.py -v
 ```
 
-**Check 2 - Hook scripts:** for each `.py` in `.claude/hooks/` (except
-`_pathfix.py`): `echo '{}' | python .claude/hooks/{hook}.py`. Exit 0 or 2,
-never 1.
+**Check 2 - Hook scripts:** for each surviving `.py` in `.claude/hooks/`
+(except `_pathfix.py`): `echo '{}' | python .claude/hooks/{hook}.py`. Exit
+0 or 2, never 1.
```
[the remark this answers is not in the document]
Comment (joshsmithxrm) [reply .claude/skills/verify/SKILL.md]: Fixed in 32b69f2a2 — Check 1 now runs the full automation suite: `python -m pytest tests/ scripts/ --import-mode=importlib -q`.

File: .claude/WORKFLOW.md [hunk 19]
```diff
@@ -57,7 +38,7 @@ Shakedown mode guarantees no real artifacts leak from exercise runs. Violations
 - `.claude/hooks/protect-main-branch.py` — also logs the raw value for debugging.
 
 ### Consumed by (skills)
-- Indirectly — skills that shell into hooks inherit the normalized value. Skills authoring new hook commands should reference `$CLAUDE_PROJECT_DIR` via forward slashes on Windows (see `/verify` `REFERENCE.md §workflow`).
+- Indirectly — skills that shell into hooks inherit the normalized value. Skills authoring new hook commands should reference `$CLAUDE_PROJECT_DIR` via forward slashes on Windows.
```
[the remark this answers is not in the document]
Comment (joshsmithxrm) [reply .claude/WORKFLOW.md]: Fixed in 32b69f2a2 — clarified that skills inherit the raw value and hooks normalize it via `_pathfix.get_project_dir()` at read time.

File: .claude/WORKFLOW.md [hunk 20]
```diff
@@ -118,30 +76,26 @@ PowerShell-native scripts (`scripts/*.ps1`) are exempt — they run under
...
 `'C:\Users\foo'` is a `SyntaxError: (unicode error) ...`.
 
 Fix the path literal at write time, not the file system at read time:
 
 | Instead of | Use |
 |------------|-----|
-| `python -c "open('.workflow\state.json')"` | `python -c "open('.workflow/state.json')"` (forward slashes) |
+| `python -c "open('specs\today.md')"` | `python -c "open('specs/today.md')"` (forward slashes) |
 | `python -c "open('C:\Users\foo')"` | `python -c "open(r'C:\Users\foo')"` (raw string) |
 | Inline heredoc + Windows path | Save to a `.py` file and run it — escapes go away |
```
[the remark this answers is not in the document]
Comment (joshsmithxrm) [reply .claude/WORKFLOW.md]: Fixed in 32b69f2a2 — the table now says moving to a .py file removes only the shell-quoting layer; forward slashes or raw strings are still required because Python parses escapes in files too.

File: CONTRIBUTING.md [hunk 21]
```diff
@@ -112,6 +112,22 @@ tests/PPDS.Cli.Tests/Services/AuthServiceTests.cs
...
+## AI-Assisted Development (Optional)
+
+This repo ships Claude Code tooling in `.claude/` — skills that document
+project procedures (release, testing conventions, multi-surface
+verification) and a small set of safety hooks (branch/key protection,
+delete guards). All of it is optional:
+
+- **CI is the contract.** Everything the tooling checks locally, CI checks
+  on your PR with plain `dotnet`/`npm` commands. You can contribute without
+  ever running an AI tool.
```
[the remark this answers is not in the document]
Comment (joshsmithxrm) [reply CONTRIBUTING.md]: Fixed in 32b69f2a2 — narrowed to 'CI is the contract for code quality' (build/test/lint parity) and identified branch/key/delete protections as local Claude Code guards, not CI checks.

File: .claude/skills/design/REFERENCE.md [hunk 22]
```diff
@@ -48,71 +38,16 @@ The fix: leave phase as `design` and let the downstream skill own its transition
...
 Runs after the bias-isolated design-fidelity review (Step 3.B). Goal: verify the spec faithfully covers the issue body — no scope drops, no reframings. This reviewer sees the issue body (non-bias-isolated by design; separate contract from Step 3.B).
 
 ### 1. Get Issue Number
 
-```bash
-python scripts/workflow-state.py get issues
-```
-
-Result is a JSON array, e.g. `[1113]`. Skip Step 3.B.2 entirely if the output is empty string (key absent), `null`, or `[]`. If the array has more than one entry, concatenate all bodies under separate `## Issue #N` headings and pass the combined text as `<ISSUE_BODY>` in a single reviewer call.
+Determine the linked issue number(s) from the branch name, the design context, or by asking the user. Skip Step 3.B.2 entirely if the design has no linked issue. If there is more than one issue, concatenate all bodies under separate `## Issue #N` headings and pass the combined text as `<ISSUE_BODY>` in a single reviewer call.
```
[the remark this answers is not in the document]
Comment (joshsmithxrm) [reply .claude/skills/design/REFERENCE.md]: Fixed in ca57e800c — §9.1 now marks `<ISSUE_BODY>` as untrusted data and the reviewer prompt mandate tells it to ignore embedded instructions and write no files (reviewer stays read-only).

File: .claude/skills/pr/SKILL.md [hunk 23]
```diff
@@ -57,49 +38,55 @@ Write body to temp file (see REFERENCE.md §3 for template), then:
...
+
 ```bash
-python scripts/workflow-state.py set pr.url "{pr-url}"
-python scripts/workflow-state.py set pr.created now
+gh pr checks <pr-number> --watch          # CI + CodeQL status
+gh pr view <pr-number> --json reviews,comments,statusCheckRollup
 ```
 
-### Step 5: Launch Background Monitor (MANDATORY) <!-- enforcement: T1 hook:session-stop-workflow -->
+Review timing is unpredictable (2–10+ minutes). Poll `gh pr view <pr-number> --json reviews,comments` every ~30s for up to **15 minutes**. If the review still has not arrived at the cap, surface that to the user and get explicit confirmation before proceeding — do not silently mark ready while a late review can still land. Skip the wait only when no reviewer app is installed on the repo.
```
[the remark this answers is not in the document]
Comment (joshsmithxrm) [reply .claude/skills/pr/SKILL.md]: Fixed in ca57e800c — Step 5 now requires evidence before skipping (reviewer config in-repo, bot activity on recent merged PRs; commands in REFERENCE.md §4) and user confirmation when inconclusive.

File: .claude/skills/pr/SKILL.md [hunk 24]
```diff
@@ -57,49 +38,55 @@ Write body to temp file (see REFERENCE.md §3 for template), then:
...
 ### Step 8: Post-Merge Cleanup
 
-Cleanup is user-initiated via `/cleanup`. See REFERENCE.md §5.
+When the PR merges, delete the branch and worktree:
+
+```bash
+git worktree remove <path>        # if working in a dedicated worktree
+git branch -d <branch>
+git push origin --delete <branch> # if the remote branch was not auto-deleted
+```
```
[the remark this answers is not in the document]
Comment (joshsmithxrm) [reply .claude/skills/pr/SKILL.md:93]: Fixed in ca57e800c — Step 8 cleanup block now opens with `gh pr view --json state` requiring MERGED, matching REFERENCE.md §5.

File: .claude/skills/qa/SKILL.md [hunk 25]
```diff
@@ -473,14 +443,10 @@ After fixing any issue found during verification:
...
 
-```bash
-python scripts/workflow-state.py set qa_findings "<JSON array of findings>"
-```
+Record each finding (PASS or FAIL) as a structured entry so the review stage can deduplicate — review will skip findings already caught and fixed by QA.
 
 Each finding entry: `{"id": "QA-N", "surface": "ext|tui|mcp|cli", "description": "...", "file": "...", "severity": "critical|important|suggestion", "fixed": true|false, "fix_commit": "<sha>|null"}`
 
-This enables the review stage to deduplicate — review will skip findings already caught and fixed by QA.
+**Handoff artifact:** write the merged findings array to `.qa-findings.json` at the repo root (an untracked working file, overwritten on each `/qa` run — do not commit it). `/review` Step 2b reads this file; if it is absent, review proceeds without dedup context.
```
[the remark this answers is not in the document]
Comment (joshsmithxrm) [reply .claude/skills/qa/SKILL.md]: Fixed in ca57e800c — schema gains `"result": "pass|fail"`, `fixed`/`fix_commit` are documented as remediation-only for fail entries, and the entries `/review` consumes for dedup are spelled out.

File: .claude/skills/shakedown/SKILL.md [hunk 26]
```diff
@@ -9,38 +9,22 @@ A structured, multi-interface product validation session. Unlike `/qa`
...
-- After modifying hooks, pipeline, or skills
-- When you want to verify the full loop: start → implement → gates →
-  verify → qa → review → converge → pr
-
 ## Phase 0: Safety Pre-checks
 
-Before any other work (either mode), validate the safety guardrails are
-configured and active.
+Before any other work, validate the safety guardrails are configured and
+active.
```
[the remark this answers is not in the document]
Comment (joshsmithxrm) [reply .claude/skills/shakedown/SKILL.md:27]: Fixed in ca57e800c — Step 3 now scopes sentinel arming to mutating Bash work; steps 1-2 are read-only checks that deliberately run first.

File: .claude/skills/verify/SKILL.md [hunk 27]
```diff
@@ -91,47 +92,13 @@ never 1.
...
-Verify is one step in the shipping pipeline, not the last one. Check
-`python scripts/workflow-state.py get phase`:
-- `implementing` -> return results to `/implement`.
-- otherwise -> proceed to `/pr` immediately. Pipeline is `/gates` ->
-  `/verify` -> `/pr`. Do not stop.
-
-Exception: on FAIL, fix first, rerun `/verify`, then `/pr`.
+Verify is one step in the usual shipping sequence (`/gates` -> `/verify`
+-> `/pr`), not the last one. On PASS, proceed to `/pr`. On FAIL, fix
+first and re-run `/verify` before continuing.
```
[the remark this answers is not in the document]
Comment (joshsmithxrm) [reply .claude/skills/verify/SKILL.md]: Fixed in ca57e800c — now reads: on FAIL run /debug to root-cause (Rule 5: verify reports, it does not fix), then re-run /verify after the fix lands.

**Comment by joshsmithxrm** [comment]
@joshsmithxrm ready for review — 6 fixed, 0 dismissed (re-review round)

File: scripts/test_skill_structure.py [hunk 28]
```diff
@@ -17,15 +17,14 @@ def _line_count(path):
...
 
-    def test_release_skill_line_count(self):
+    def test_release_skill_files_exist(self):
+        # PR #1321 deliberately restored the full /release runbook into
+        # SKILL.md (far past the 150-line two-file cap), making release a
+        # standing exception to the cap. Assert only that both files exist.
         path = SKILLS / 'release' / 'SKILL.md'
         ref = SKILLS / 'release' / 'REFERENCE.md'
         self.assertTrue(path.exists())
         self.assertTrue(ref.exists())
```
[the remark this answers is not in the document]
Comment (joshsmithxrm) [reply scripts/test_skill_structure.py]: Fixed in d3f05ff06 — assert is_file() instead of exists() so a directory named SKILL.md/REFERENCE.md can't satisfy the check.
